saq/worker.py

```diff
@@ -140,32 +142,34 @@ async def monitor(self, task, job_id):
                 await job.update()
 
     async def process(self):
+        # pylint: disable=too-many-branches
         job = None
         monitor = None
 
         try:
-            job = await self.queue.dequeue()
-            job_id = job.id
-            job.started = now()
-            job.status = Status.ACTIVE
-            job.attempts += 1
-            logger.info("Processing %s", job)
-            await job.update()
-            context = {**self.context, "job": job}
+            job = await self.queue.dequeue(self.dequeue_timeout)
```

Comment:
can we do an early return here instead?

we don't care about any call backs if there's no job right